Add Documentation - Framework - Plugin Context Accessors (#1883)
kuzzleio/kuzzle: doc/2/framework/classes/plugin-storage/bootstrap/index.md

---
code: true
type: page
title: bootstrap
description: PluginStorage class bootstrap() method
---

# bootstrap

Initializes the plugin private storage.

> Can be called any number of times as long as identical mappings are provided.

Reviewer: ```suggestion
Can be called ad much as needed as long as identical mappings are provided.
```

## Arguments

```ts
bootstrap(collections: JSONObject) => Promise<void>;
```

<br/>

| Arguments     | Type              | Description                                                                                                                        |
| ------------- | ----------------- | ---------------------------------------------------------------------------------------------------------------------------------- |

> | `collections` | <pre>JSONObject</pre> | List of collection to create, with their corresponding [mappings](/core/2/guides/maint-concepts/2-data-storage#collection-mappings) |

Reviewer 2: ```suggestion
| `collections` | <pre>JSONObject</pre> | List of collections to create, with their corresponding [mappings](/core/2/guides/maint-concepts/2-data-storage#collection-mappings) |
```
